### ISSUE TITLE
HTML that contain @ cause stray "–>" in output

### ISSUE THREAD

**Issue by amartini51** [apple/swift-docc#409]
### Description

Input like the following causes the issue:

```
Hi there.

<!--
  @test
-->
```

<img width="123" alt="image" src="https://user-images.githubusercontent.com/15617351/195960895-5adcfcd7-04b6-4d19-816a-6c7174fec634.png">

In contrast, this renders correctly:

```
Hi there.

<!--
  test
-->
```

<img width="96" alt="image" src="https://user-images.githubusercontent.com/15617351/195960904-765d28cf-792d-4d20-ae8e-b767dc970a5d.png">

It seems like DocC is trying to parse an `@` directive block in the middle of the comment.

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc/issues).

**Comment by ethan-kusters** [comment]
@QuietMisdreavus should we transfer this to [swift-markdown](https://github.com/apple/swift-markdown)?

**Comment by QuietMisdreavus** [comment]
Yes please! The bug would be solved there. I'll do the transfer.
